lib/helper/update_manager.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:smarthome/helper/preference_manager.dart';
+import 'package:smarthome/helper/simple_dialog.dart' as simpleDialog;
+import 'package:smarthome/models/versionAndUrl.dart';
+import 'package:version/version.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+
+class UpdateManager {
+  static final Version version = Version(1, 1, 5);
+  static final int checkEveryHours = 16;
+  static final GitHub gitHub = GitHub();
+  static final RepositorySlug repositorySlug = RepositorySlug("susch19", "SmartHome");
+  static final RegExp versionRegExp = RegExp(r'v|V');
+  static DateTime? lastChecked;
+
+  static final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin();
+
+  static const String updateNotificationTitle = "Neue Version verfügbar"; // Update available
+  static const String updateNotificationBody =
+      "Bitte aktualisiere auf "; // A new update is available. Please update to version
+
+  static int notificationId = 0;
+
+  static Future<void> initialize() async {
+    lastChecked = PreferencesManager.instance.getDateTime("lastChecked");
+
+    const AndroidInitializationSettings initializationSettingsAndroid =
+        AndroidInitializationSettings('@mipmap/ic_launcher');
+
+    final InitializationSettings initializationSettings =
+        InitializationSettings(android: initializationSettingsAndroid);
+
+    await flutterLocalNotificationsPlugin.initialize(initializationSettings, onSelectNotification: (String? payload) {
+      if (payload != null) {
+        HelperMethods.openUrl(payload);
+      }
+    });
+  }
+
+  static Future<void> showUpdateNotification(BuildContext context) async {
+    print("showUpdateNotification called " + (lastChecked?.toString() ?? "lastChecked is null"));
+    // TODO: at a later point maybe schedule this notification to be shown every x hours
+    // TODO: maybe also put the check every x hours in the settings
+    if (lastChecked == null || DateTime.now().difference(lastChecked!).inHours > checkEveryHours) {
+      lastChecked = DateTime.now();
+      PreferencesManager.instance.setDateTime("lastChecked", lastChecked!);
+
+      final versionAndUrl = await getVersionAndUrl();
+      if (versionAndUrl?.url != null && await _isNewVersionAvailable(versionAndUrl?.version)) {
+        if (Platform.isAndroid) {
+          // Show notification
+          await _showNotification(versionAndUrl!);
+        } else {
+          // Show dialog
+          showDialog(
+              context: context,
+              builder: (BuildContext c) => simpleDialog.SimpleDialog.create(
+                  context: c,
+                  title: updateNotificationTitle,
+                  content: updateNotificationBody + versionAndUrl!.version.toString(),
+                  okButtonText: "Aktualisieren",
+                  cancelButtonText: "Später",
+                  onSubmitted: () => HelperMethods.openUrl(versionAndUrl.url!)));
```

Comment:
```suggestion
                  onSubmitted: () => HelperMethods.openUrl(versionAndUrl.url)));
```

Not nullable with the rest of my suggestions